lib/environment-variables-manager/environment-variables-manager.ts

```diff
@@ -1,102 +1,27 @@
-import * as process from 'process';
-
-import { JsonValue } from 'lib/types/general';
-import {
-  EnvironmentData,
-  GetOptions,
-  IEnvironmentVariablesManager,
-  Options
-} from 'types/environment-variables-manager';
-import { isDefined } from 'types/guards';
-import { isLocalEnvironment } from 'utils/env';
-import { Logger } from 'utils/logger';
+import { KeyValueManager } from 'lib/key-value-manager/key-value-manager.abstract';
+import { Options } from 'types/key-value-manager';
 import { snakeCase } from 'utils/string-manipulations';
 
-const logger = new Logger('EnvironmentVariablesManager', { mondayInternal: true });
-
-const readEnvironmentData = () => {
-  const environmentDataFileName = process?.env?.SECRET_NAME;
-  if (!isDefined(environmentDataFileName)) {
-    logger.error('[EnvironmentVariablesManager] Environment file name is not defined');
-    return;
-  }
-  
-  const environmentDataFilePath = `/secrets/${environmentDataFileName}`;
-  const isEnvironmentFileExists = fs.existsSync(environmentDataFilePath);
-  if (!isEnvironmentFileExists) {
-    logger.error('[EnvironmentVariablesManager] Environment file does not exist in path');
-    return;
-  }
-  
-  const environmentData = fs.readFileSync(environmentDataFilePath, 'utf8');
-  try {
-    return JSON.parse(environmentData) as EnvironmentData;
-  } catch (error) {
-    logger.error('[EnvironmentVariablesManager] Corrupted Environment file. File is not in JSON format');
-    return;
-  }
-};
+export class EnvironmentVariablesManager extends KeyValueManager {
+  private readonly shouldUpdateProcessEnv: boolean;
 
-export class EnvironmentVariablesManager implements IEnvironmentVariablesManager {
-  private cachedEnvironmentData?: EnvironmentData;
-  private shouldUpdateProcessEnv: boolean;
-  
   constructor(options?: Options) {
-    this.shouldUpdateProcessEnv = !!options?.updateProcessEnv;
-    this.initEnv(options);
-    if (isLocalEnvironment()) {
-      logger.info('[EnvironmentVariablesManager] Running in development environment, Using process.env', { mondayInternal: false });
-    }
-  }
-  
-  private initEnvIfNeeded(options?: GetOptions) {
-    const shouldInvalidate = options?.invalidate ?? true;
-    if (shouldInvalidate || !this.cachedEnvironmentData) {
-      this.initEnv();
-    }
+    const dataFileName = process.env.SECRET_NAME!;
+    super('EnvironmentVariablesManager', dataFileName, 'secrets');
+    this.shouldUpdateProcessEnv = Boolean(options?.updateProcessEnv);
+    this.initData();
   }
-  
-  private initEnv(options: Options = {}) {
-    if (isLocalEnvironment()) {
-      this.cachedEnvironmentData = process.env as EnvironmentData;
-      return;
-    }
-    
-    const { updateProcessEnv } = options;
-    this.shouldUpdateProcessEnv = updateProcessEnv ?? this.shouldUpdateProcessEnv;
-    this.cachedEnvironmentData = readEnvironmentData();
-    if (!isDefined(this.cachedEnvironmentData)) {
-      return;
-    }
-    
+
+  protected initData() {
```

Comment:
why is this protected? should be private, right?

Reply:
no. it is called explictly from the constructors of the implementing classes because the env-var-manager has further logic that extends the base class.
so it is protected so it will be available in child classes.

env-var-manager:
```
  protected initData() {
    super.initData();

    if (this.shouldUpdateProcessEnv) {
      Object.entries(this.cachedData!).forEach(([key, value]) => {
        const snakeCaseKey = snakeCase(key, { upperCase: true });
        process.env[snakeCaseKey] = value as string;
      });
    }

    this.logger.info(`[${this.tag}] Successfully updated process.env with data from file`, { mondayInternal: true });
  }
```